rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,210 @@
+to share the same *Major* and *Minor* version numbers. Each project is
+going to have an independent *Patch* level number.
+
+All the non-visible components of Kubewarden (e.g. the different
+Rust libraries) and the policies are going to retain their
+existing version numbers, release strategy and, generally speaking,
+independence.
+
+### Helm charts
+
+Helm charts have two kind of version numbers:
+
+* `version`: a SemVer 2 version specific to the helm chart
+* `appVersion`: the version of the app that the chart contains
+
+The helm charts will keep their own independence when it comes to
+the `version` attribute.
+On the other hand, the `appVersion` attribute
+will always be set to the `<Major>.<Minor>` version of the
+Kubewarden stack.
+
+> See the official documentation about
+[`Chart.yaml`](https://helm.sh/docs/topics/charts/#the-chartyaml-file)
+for more information.
+
+## Examples
+
+This section provides clarity about how the proposal would work
+using some concrete real-world scenarios.
+
+### A new release Kubewarden takes place
+
+A new version of Kubewarden stack has to be released. That can
+happen because a new important feature/bugfix has been added,
+because of marketing reasons,...
+
+Let's assume:
+
+* The current version of the Kubewarden stack is `0.5.3`
+* The current version of the helm chart is `v0.10.3`
+
+The new feature is not worth a major bump of the "unique kubewarden stack version",
+hence we will perform a minor update.
+
+We are going to tag `v0.6.0` of all these components and release them:
+
+* Kubewarden Controller
+* Policy Server
+* kwctl
+
+This will happen **regardless** of the changes done to these projects. For
+example, kwctl is going to have a new release `v0.6.0` even if it didn't
+have any commits since `v0.5.3` was tagged.
+
+The Helm charts are updated in this way:
+
+* The version of container images (Policy Server, Kubewarden
+  Controller) are set to be `0.6.0`
+* The chart `version` attribute is set to `v0.10.4` because no significant
+  changed happened
+* The `appVersion` attribute is set to `0.6.0`
+
+### A patch is done to kwctl
+
+A series of minor issues are found inside of kwctl. We decide to
+tag a new release that includes these fixes.
+
+The version of `kwctl` is updated from `0.6.0` to `0.6.1`.
+No other releases are done
+
+### A patch is done to Kubewarden Controller
+
+A minor issue is fixed with Kubewarden Controller. We decide to
+tag a new release that includes the fix.
+
+The following actions are done:
+
+* We tag a new release of kubewarden-controller: `0.6.1`
+* We update the helm chart that makes use of this image:
+  * We update the references of the image to ensure `0.6.1` is being pulled
+  * We update the `version` number of the chart itself to be
+    change from `0.10.4` to `0.10.5`.
+    
+    **Note:** we don't change the `appVersion` attribute
+
+### A major security issue is found inside of Policy Server
```

Comment:
An interesting second take of this one is:

**A major security issue is found inside of Policy Server**: if the controller does not need any adaptation, only the Policy Server and the default Helm chart have to be updated.

An open question in this case is how we are going to bump the existing policy-server images: this is a problem regardless of the single version/umbrella versions proposal though.

Reply:
Added